File: app-java/src/main/java/com/java/bilibili/ui/main/fragment/adapter/HomePagerAdapter.java
package com.java.bilibili.ui.main.fragment.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;



public class HomePagerAdapter extends FragmentPagerAdapter {
    private String[] mTitle;
    private List<Fragment> fragments;

    public HomePagerAdapter(FragmentManager fm, List<Fragment> fragments, String[] title) {
        super(fm);
        this.fragments = fragments;
        this.mTitle = title;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitle[position];
    }

}

File: app-java/src/main/java/com/java/bilibili/ui/main/fragment/adapter/HomePagerAdapter.kt
package com.kotlin.bilibili.ui.main.fragment.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

class HomePagerAdapter(fm: FragmentManager, private val fragments: List<Fragment>, private val mTitle: Array<String>) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return mTitle[position]
    }
}
